### ISSUE TITLE
[ENH?]: EngFormatter: How can I override ENG_PREFIXES?

### ISSUE THREAD

**Issue by ishandutta2007** [matplotlib/matplotlib#28516]
### Problem

The hard-coded ENG_PREFIXES looks like this:
```
    ENG_PREFIXES = {
        -24: "y",
        -21: "z",
        -18: "a",
        -15: "f",
        -12: "p",
         -9: "n",
         -6: "\u03bc",
         -3: "m",
          0: "",
          3: "k",
          6: "M",
          9: "G",
         12: "T",
         15: "P",
         18: "E",
         21: "Z",
         24: "Y"
    }
```
For value corresponding to `9` is  "G", I need it to be "B" instead.

**Comment by timhoffm** [comment]
Matplotlib does not intend to make this configurable. You have two options:

- Quick-and-dirty: Monkey-patch `ENG_PREFIXES`. - You have to judge whether that's acceptable in your context.
- Write your own formatter:
  ```
  class CustomEngFormatter(EngFormatter):
      ENG_PREFIXES = {
          ...
      }
  ```

**Comment by ishandutta2007** [comment]
@timhoffm 

Yes by override I meant the 2nd option but instead of copy pasting all functions into a totally new class it should be more like `class CustomEngFormatter(EngFormatter) extends EngFormatter`


And can you share the thread where and why it was decided to not make it configurable ?

**Comment by timhoffm** [comment]
I'm not following. `class CustomEngFormatter(EngFormatter):` inherits all functionality from `EngFormatter`. You only have to specify a new set of prefixes.

There is no related discussion. I've grounded my claim on the fact that the prefixes are a [standard convention](https://en.wikipedia.org/wiki/Metric_prefix) and not something people should modify under normal circumstances. As API lead I would thus oppose making this configurable. For the very rare cases that you really need something else, you have the two options I've proposed above.
